core/template.php

```diff
@@ -244,67 +249,76 @@ function tpl_list($list)
 {
     global $smarty, $config;
     global $listcolumns;
-
```

Comment:
a lot of unrelated changes?

Reply:
count didnt work on null or undefined list in php 7.2 you get a warning see https://www.php.net/manual/function.count.php so you got an warning on the search-start site

Reply:
I feel it would be cleaner to:

    $smarty->assign('totalresults', 0);

and then return

    if (!$list) { return }

Then you avoid the additional level of indents and checking

Reply:
changed